pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader.go

```diff
@@ -22,19 +23,13 @@ import (
 
 var _ types.ChainReader = (*Client)(nil)
 
-// NewChainReaderTestClient is a test client for [types.ChainReader]
-// internal users should instantiate a client directly and set all private fields.
-func NewChainReaderTestClient(conn *grpc.ClientConn) types.ChainReader {
-	return &Client{grpc: pb.NewChainReaderClient(conn)}
-}
-
 type Client struct {
-	*net.BrokerExt
+	*goplugin.ServiceClient
```

Comment:
why did this change? i'm surprised you can make this change without effecting the client connectivity